### PR TITLE
Prac 06 feedback

### PR DESCRIPTION
**Pull request by JarrodE**
Hello @Nathan-Dachille Could you please review my code for week 6! I tried to commit more often this week but i'm unsure if I worded my commits correctly. Cheers!

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def __str__(self):  # 5 Add the __str__ method
+        """Return a string version of car object"""
```
Comment (Nathan-Dachille) [inline prac_06/car.py:35]: Nit: To maintain consistancy remember to end docstrings with a full stop.

File: prac_06/car.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return f"{self.name}, fuel={self.fuel}, odometer={self.odometer}"  # Task 6
```
Comment (Nathan-Dachille) [inline prac_06/car.py:36]: FYI: This file is missing a newline at the end, the hotkey <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>L</kbd> can be used in PyCharm to automatically format your code.

File: prac_06/guitars.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from guitar import Guitar
+
+
+def main():
+    """Main program to store and display guitar details."""
+    guitars = get_user_guitars()
+    display_guitars(guitars)
+
+
+def get_user_guitars():
```
Comment (Nathan-Dachille) [inline prac_06/guitars.py:16]: What does this function do, perhaps a docstring could help?

File: prac_06/guitars.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        year = int(input("Year: "))
+        cost = float(input("Cost: $"))
+        new_guitar = Guitar(name, year, cost)
+        guitars.append(new_guitar)
+        print(f"{new_guitar} added.")
+        name = input("Name: ")
+    return guitars
+
+
+def display_guitars(guitars):
```
Comment (Nathan-Dachille) [inline prac_06/guitars.py:30]: Remember to add docstrings for all functions.

File: prac_06/programming_language.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
+        """Return true if the programming language is dynamically typed."""
+        return self.typing == "Dynamic"
+
+    def __str__(self):
+        """Print python string"""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, First appeared in {self.year}"
```
Comment (Nathan-Dachille) [inline prac_06/programming_language.py:25]: Nit: Method does not print a string, however it does return one.

File: prac_06/programming_language.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Estimated: 30 minutes
+Actual: 21 minutes
+Current time: 16:54
+Finish time: 17:15
+"""
+
+
+class ProgrammingLanguage:
+    """Represent a Programming Language object."""
+
```
Comment (Nathan-Dachille) [inline prac_06/programming_language.py:11]: Docstrings shouldn't have a blank line before or after them.

**Review by Nathan-Dachille — APPROVED** [review]
Good evening @JarrodE,

Most of the feedback that I have provided is mostly just nitpicking, things such as missing full stops in docstrings or forgetting to leave a blank line at the end of your programs. If you want to guarantee that the file is correctly formatted in PyCharm you can use the hotkey <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>L</kbd>. Other then that your commit messages are excellent.

Cheers, Nathan.

**Comment by JarrodE** [comment]
Awesome! Thanks @Nathan-Dachille - or some reason I always struggle with docstrings, but getting better! Thanks!

**Pull request merged**